components/metadata/metadata_create/operator_create/operator_create.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/python
+import argparse
+import logging
+import os
+import sys
+import requests
+import json
+import base64
+import uuid
+
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "..", "tokens"))
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "..", "utils"))
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "..", "metadata"))
+
+from get_token import *
+from utils import *
+from metadata_traits import *
+
+session = requests.Session()
+
+def main():
+    # >> Arguments
+    arg_parser = argparse.ArgumentParser(description="Upload an Operator.")
+
+    # script parameters:
+    arg_parser.add_argument("--log-format", default='> %(asctime)-15s %(module)s %(levelname)s %(funcName)s:   %(message)s')
+    arg_parser.add_argument("--log-level", default=logging.INFO)
+
+    # server parameters:
+    arg_parser.add_argument("--dc", default="", required=True)
+    arg_parser.add_argument("--env", default="", help="deploy environment (which determines server location)")
+    arg_parser.add_argument("--oauth_id", default="", help="oauth_id")
+    arg_parser.add_argument("--oauth_secret", default="", help="oauth_secret")
+    arg_parser.add_argument("--oauth_scope", default="", help="oauth_scope")
+
+    # request parameters:
+    arg_parser.add_argument("--site_id", default="", help="Site Identifier", required=True)
+    arg_parser.add_argument("--operator_first_name", default="", help="The first name of the operator", required=True)
+    arg_parser.add_argument("--operator_last_name", default=None, help="The last name of the operator", required=True)
+    arg_parser.add_argument("--operator_code", default=None, help="An optional code to describe the operator")
+
+    arg_parser.set_defaults()
+    args = arg_parser.parse_args()
+    logging.basicConfig(format=args.log_format, level=args.log_level)
+    # << Arguments
+
+    server = ServerConfig(a_environment=args.env, a_data_center=args.dc)
+   
+    logging.info("Running {0} for server={1} dc={2} site={3}".format(os.path.basename(os.path.realpath(__file__)), server.to_url(), args.dc, args.site_id))
+
+    token = get_token(a_client_id=args.oauth_id, a_client_secret=args.oauth_secret, a_scope=args.oauth_scope, a_server_config=server)
+
+    operator_rdm_bean = OperatorMetadataTraits.post_bean_json(a_first_name=args.operator_first_name, a_last_name=args.operator_last_name, a_code=args.operator_code)

```

Comment:
Move this to separate function to match the pattern you've used elsewhere?

Reply:
Good find.